File: tests/fixtures/typescript/sample-repo/src/randomness.ts
// Random-effect patterns.

import { randomBytes, randomUUID } from "node:crypto";

export function rollDie(): number {
  // expect: Random=definite
  return Math.floor(Math.random() * 6) + 1;
}

export function newSessionId(): string {
  // expect: Random=definite
  return randomUUID();
}

export function nonce(length: number): Buffer {
  // expect: Random=definite
  return randomBytes(length);
}

export function shuffleInPlace<T>(items: T[]): void {
  // expect: Random=definite, Mut=definite
  for (let i = items.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [items[i], items[j]] = [items[j]!, items[i]!];
  }
}

export function fakeRandom(): number {
  // expect: Random=probable
  // Method `.random()` on an unresolved object.
  const bag = { random: () => 4 };
  return bag.random();
}

File: tests/fixtures/rust/sample-repo/src/randomness.rs
// Random-effect patterns.

use rand::Rng;
use uuid::Uuid;

pub fn roll_die() -> u8 {
    // expect: Random=definite
    rand::thread_rng().gen_range(1..=6)
}

pub fn new_session_id() -> Uuid {
    // expect: Random=definite
    Uuid::new_v4()
}

pub fn shuffle_in_place<T>(items: &mut [T]) {
    // expect: Random=definite, Mut=definite
    use rand::seq::SliceRandom;
    items.shuffle(&mut rand::thread_rng());
}

pub fn fixed_v5(name: &str) -> Uuid {
    // expect: (none — Pure; v5 is deterministic)
    Uuid::new_v5(&Uuid::NAMESPACE_DNS, name.as_bytes())
}
